### PR TITLE
feat: DAMD-146 (add proposal_id and ob_code to pfsDesign/pfsConfig fi…

### PR DESCRIPTION
**Pull request by monodera**
…les)

- `proposalId` and `obCode` are added.
- `datamodel.txt` and `test_PfsConfig.py` are modified accordingly.

### CODE REVIEW

File: datamodel.txt [hunk 1]
```diff
@@ -385,6 +385,8 @@ The DESIGN table lists for each object:
       targetType     32-bit int (enumerated type: e.g. SCIENCE,SKY,FLUXSTD)
       fiberStatus    32-bit int (enumerated type: e.g. GOOD,BROKENFIBER,BLOCKED,BLACKSPOT)
       pfiNominal     pair of 32-bit floats (millimeters on the PFI)
+      proposalId     string
+      obCode         string
```
Comment (PaulPrice) [inline datamodel.txt:389]: Please add more explanation of these below.

File: python/pfs/datamodel/pfsConfig.py [hunk 2]
```diff
@@ -135,6 +135,10 @@ class PfsDesign:
     fiberStatus : `numpy.ndarray` of `int`
         Status of each fiber. Values must be convertible to `FiberStatus`
         (which limits the range of values).
+    proposalId : `numpy.chararray`
+        Proposal ID of each fiber (e.g, S23A-001QN).
+    obCode : `numpy.chararray`
+        OB code of each fiber.
```
Comment (PaulPrice) [inline python/pfs/datamodel/pfsConfig.py:141]: Spell out OB: "Observing block".

**Review by PaulPrice — CHANGES_REQUESTED** [review]
`test_PfsFluxReference.py` fails due to this change, and I'm sure there are similar tests in drp_stella that fail too.

**Comment by PaulPrice** [comment]
Please squash related changes, and use `git rebase` instead of merging `master` into the ticket branch.

**Comment by monodera** [comment]
I think that I have squashed everything suggested by `git rebase` in this and the other related repositories. I've also run the integration test locally and succeeded.

**Comment by PaulPrice** [comment]
The current state is a bit of a mess, with multiple commits with the same title.

**Comment by monodera** [comment]
What can I do from now to clean up thing?

**Comment by PaulPrice** [comment]
If you're at NAOJ, perhaps @SogoMineo is able to help guide you through the git labyrinth?

**Comment by monodera** [comment]
we are in different branches, but I'll ask him.

**Comment by monodera** [comment]
With the great help of @SogoMineo, I restarted editing from the current master branch and force-pushed the commits as shown in the tree. 

A few places differ from the older commits, but these lines are duplicates from my wrong edit in the previous rebase.  You can expand the code and find the identical lines a few lines below the removed ones.

**Review by PaulPrice — APPROVED** [review]
Very nice. Thank you, @SogoMineo! And thank you @monodera for bearing with me.

**Comment by monodera** [comment]
Can I merge to the master or you will merge, @PaulPrice ?

**Comment by monodera** [comment]
@SogoMineo told me that I can push the green "merge pull request" button as encouraged in the documentation, so I'll do it.

**Comment by monodera** [comment]
@PaulPrice When merging a PR, which method should I use from the menu, "Create a merge commit", "Squash and merge", and "Rebase and merge"?

Also, could you please take a look at these related branches in `drp_stella` and `pfs_utils`?
https://github.com/Subaru-PFS/drp_stella/pull/330
https://github.com/Subaru-PFS/pfs_utils/pull/65

**Comment by PaulPrice** [comment]
Please use the "Create a merge commit" method.
The others look fine too.
You've run these through the integration test, right?

**Comment by monodera** [comment]
Yes, it finished with the "Done with Gen3." message.

**Pull request merged**